### DISCUSSION TITLE
Convenient ways to pass a supergraph to the router

### DISCUSSION THREAD

**Discussion by o0Ignition0o** [apollographql/router#468] [Implementation]
This is related to #377 where I was trying to pass a -u url to a supergraph for demo / convenience purposes.

However this might not be the right way to reason about it.

Other approaches may include piping a supergraph to stdin, or let rover provide this mechanism (cc @EverlastingBugstopper )

 Quoting @abernix [comment](https://github.com/apollographql/router/pull/377#issuecomment-1027745873) "like allowing --supergraph to accept STDIN (and thus enabling something like curl -ssL https://go.apollo.dev/r/supergraph | ./router -s -)."

**Comment by EverlastingBugstopper** [comment]
from stdin seems reasonable to start. you could do something like `rover supergraph compose --config supergraph.yml > ./router -s -`

we'll eventually want a way to watch for file changes and recompose whenever one of the subgraph sources updates, so something like `rover supergraph compose --config supergraph.yml --watch --outfile supergraph.graphql` would be something you could run and then you could have the router watch _that_ output file with something like `router --watch -s supergraph.graphql`.

down the line we could have some other mechanism entirely, like maybe we have `rover supergraph dev` watches subgraph sources and recomposes them for you, which would then send that new schema over to the router process. not sure exactly how that would look under the hood but i could imagine we have rover start up the router with the first composed schema, then open a webhook in the router that listens for future schemas which rover would then push to it when it can. that might end up being overkill but maybe not. also maybe there's another better way i haven't thought of, but the live reload mechanism is definitely something we'll want to have. (other interesting questions arise like, should we also try to handle live reloading the subgraphs themselves if something changes, but this seems difficult to do in a cross platform way)
